File: examples/clojure/features/step_definitions/calculator_steps.clj
(use 'calculator)
;; We're not using test-is yet, since failures are just printed - nothing is raised.
;; (use 'clojure.contrib.test-is)

;; Just test that it works
(After)

(Before
  (dosync (ref-set numbers [])))

(Given #"^I have entered ([\d.]+) into the calculator$"
  #(push-number (Float. %1)))

(When #"^I press (\w+)$"
  #(calculate ({"divide" / "add" +} %)))

;; Verbose style
(Then #"^the current value should be ([\d.]+)$"
  (fn [expected]
;;    (is (= (current-value) (Float. expected)))))
    (assert (= (current-value) (Float. expected)))))

File: examples/javascript/features/step_definitions/calculator_steps.js
load('lib/calculator.js');

Given(/^I have entered (\d+) into the calculator$/, function(n) {
    calculator.push(parseFloat(n));
});

When(/^I press (\w+)$/, function(button) {
    this.top = calculator.divide();
});

Then(/^the current value should be (.*)$/, function(value) {
    value = parseFloat(value);
    if (this.top != value) {
        throw "Expected " + value + ", but got " + this.top;
    }
});
